app/models/setting/remote_execution.rb

```diff
@@ -73,6 +73,12 @@ def self.default_settings
         _('Form Job Template'),
         nil,
         { :collection => proc { Hash[JobTemplate.unscoped.map { |template| [template.name, template.name] }] } }),
+      self.set('remote_execution_job_invocation_template',
+        N_('Select a report template used for generating a report for a particular remote execution job'),
+        'Job Invocation report',
+        _('Job Invocation Report Template'),
+        nil,
+        { :collection => proc { Hash[ReportTemplate.unscoped.map { |template| [template.name, template.name] }] } }),
```

Comment:
Should we help the user by? We require the input to be present, so it would make sense to offer only templates that has it.
```suggestion
        { :collection => proc { Hash[ReportTemplate.unscoped.join(:template_inputs).where(TemplateInput.arel_table[:name].eq('job_id')).map { |template| [template.name, template.name] }] } }),
```

Reply:
Great idea!